rtsf-at-checkout-device-scale/driver/scaleDevice_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright © 2022 Intel Corporation. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+package driver
+
+import (
+	"device-scale/scale"
+	"reflect"
+	"testing"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_scaleDevice_readWeight(t *testing.T) {
+
+	config := scale.Config{
+		PortName:        "/dev/tty.usbserial-test",
+		BaudRate:        9600,
+		DataBits:        7,
+		StopBits:        1,
+		MinimumReadSize: 1,
+		ParityMode:      2,
+		TimeOutMilli:    500,
+	}
+
+	testDevice := scale.InitializeMockDevice(&config)
+	tests := []struct {
+		name          string
+		serialDevice  *scale.MockDevice
```

Comment:
No needed if it is the same for all the tests.

Reply:
@lenny-intel I need it to run the serialDevice's functions, such as getReading